src/twisted/web/server.py

```diff
@@ -500,6 +505,27 @@ def getRootURL(self):
         return self.appRootURL
 
 
+    def _handleStar(self):
+        """
+        Handle receiving a request whose path is '*'.
+
+        RFC 7231 defines an OPTIONS * request as being something that a client
+        can send as a low-effort way to probe server capabilities or readiness.
+        Rather than bother the user with this, we simply fast-path it back to
+        an empty 200 OK. Any non-OPTIONS verb gets a 405 Method Not Allowed
+        telling the client they can only use OPTIONS.
+        """
+        if self.method == b'OPTIONS':
+            self.setResponseCode(http.OK)
+        else:
+            self.setResponseCode(http.NOT_ALLOWED)
+            self.setHeader(b'Allow', b'OPTIONS')
+
```

Comment:
I dunno… 400 seems pretty correct:

> something that is perceived to be  a client error

`*` as a URL is clearly a client error.

As to whether the user should get a look, I have to disagree.  This is rare enough to fall into the "almost certainly don't care" category, but an application should always have a hook into handling the request if they want to.

A lot of applications choose to explain the error in the body with some HTML or text, for example.  Perhaps you want to redirect bad requests to [YouTube](https://www.youtube.com/watch?v=dQw4w9WgXcQ).  Or you're that guy that sends non-ASCII header names and also use `*` as a magic thing.  You know.